testsuite/tests/iterators/big_iterator/test.py

```diff
@@ -0,0 +1,38 @@
+"""
+Test iterating over a huge number of elements.
+
+This shows that an iterator should be used in place of an array when one wants
+to return a huge number of elements, as calling ``p_get_all_examples`` from
+the public APIs crashes in this instance.
```

Comment:
When possible, I think we should avoid committing huge files (this is for practical reasons: keep the git repo small to keep it convenient to handle).

Here, we just need a big array in the DSL. A huge list of parsing nodes is one way to do it (with this huge file con), but another way would be to have a property that takes an array and that returns an iterator over that array, and do from the Python script: `u.root.p_iter_array(range(10**9))` (likewise in Ada)